src/net.cpp

```diff
@@ -1097,10 +1097,13 @@ void ThreadDNSAddressSeed()
         } else {
             vector<CNetAddr> vIPs;
             vector<CAddress> vAdd;
-            if (LookupHost(seed.host.c_str(), vIPs))
+            if (LookupHost(((seed.fXoredIPs ? "x." : "") + seed.host).c_str(), vIPs))
             {
                 BOOST_FOREACH(CNetAddr& ip, vIPs)
                 {
+                    if (seed.fXoredIPs)
+                        ip.XORAddress();
```

Comment:
Hmm... i thought about this when i was writing to code.
But because the XORed ip is kind of a cyphertext, i was thinking a in-state manipulating of the instance variable `ip` is appropriate. But i also dislike state changing functions for such cases.